### PR TITLE
Add csv driver option to specify csv read args

### PR DESCRIPTION
**Pull request by MaxSchambach**
# Description

Adds a `read_csv_kwargs` argument to `CsvDriver` initialization which is used in all `read_csv` calls in the class.
Does not break backward compatibility, as `get_df` and `get_iter` still allow to specify kwargs for `read_csv` which will take precedence over the ones given at initialization.

This makes the creation of new catalog entries based on the `CsvDriver` much easier as dataset-specific read options (such as seperator, dtypes, etc.) can be specified in the `driver_kwargs`.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [x] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [x] Lint and unit tests pass locally with my changes
- [x] I have kept the PR small so that it can be easily reviewed  
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -462,62 +462,6 @@ google-resumable-media==2.3.3 \
 googleapis-common-protos==1.56.4 \
     --hash=sha256:8eb2cbc91b69feaf23e32452a7ae60e791e09967d81d4fcc7fc388182d1bd394 \
     --hash=sha256:c25873c47279387cfdcbdafa36149887901d36202cb645a0e4f29686bf6e4417
-greenlet==1.1.2 \
```
Comment (pzdkn) [inline requirements.txt:465]: doesn't a change here mean that `requirements.in` also got changed?

Comment (MaxSchambach) [reply requirements.txt:465]: Dang, I didn't do any requirements changes and did not commit that on purpose. Will revert.

Comment (MaxSchambach) [reply requirements.txt]: Done, thanks for the catch!

**Comment by MaxSchambach** [comment]
> I don't understand, why the test fails due to changes introduced by `black`. I did run `black` locally before submitting and don't see why it fails in the cloud test. Any ideas @pzdkn ?

should be solved now. weirdly, `pre-commit` performed a different `black` call than calling `black` directly.

**Review by pzdkn — APPROVED** [review]
Thanks Max, LGTM!

**Pull request merged**